<launch>

	<param name="use_sim_time" value="false" />

	<arg name="launch_prefix" default="" />
	<!-- set to value="gdbserver localhost:10000" for remote debugging -->

	<!-- Set parameters -->


	<node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)">
		<rosparam command="load" file="$(env HOME)/dmnaac_ws/apriltag_landmarklist_ros/src/apriltag_ros/config/settings.yaml" />
		<rosparam command="load" file="$(env HOME)/dmnaac_ws/apriltag_landmarklist_ros/src/apriltag_ros/config/tags.yaml" />

		<!-- <param name="camera_frame" type="str" value="usb_camera_L" /> -->
		<param name="publish_tag_detections_image" type="bool" value="false" />
		<!-- <remap from="/tag_detections_image" to="/tag_detections_image_L" />
			<remap from="/tag_detections" to="/tag_detections_L" /> -->
		<remap from="image_rect" to="/cam_L/image_raw" />
		<remap from="camera_info" to="/cam_L/camera_info" />
		<!-- the following param cannot be dynamic changed as it only be read by init-->
		<!-- <param name="gray_threshold" type="int" value="1" />	added in the library: continous node -->
		<!-- 0: No threshold 1: All use threshold 2: switch grey and threshold  -->
		<!-- <param name="use_gray_mode" type="int" value="0" /> -->

		<!-- Parameters for cartographer_ros landmarks -->
		<param name="publish_landmarks" type="bool" value="true" />
		<param name="translation_weight" type="double" value="100000.0" />
		<param name="rotation_weight" type="double" value="100000.0" />
		<param name="tracking_frame" type="string" value="base_footprint" />

		<!-- Parameters for saving poses of tags relative to map_frame -->
		<param name="map_frame" type="string" value="map" />
		<param name="enable_write_tags" type="bool" value="true" />

		<!-- Path to the pose.txt file which is created when calling the write_tags service.
		This parameter is used only when enable_write_tags is set to false. -->
		<param name="path_to_pose_txt" type="string" value="" />
		<param name="publish_robot_pose" type="bool" value="true" />

	</node>



</launch>


